File: PumpLogUI/src/models/section.ts
export interface Session {
  sessionGuid: string;
  sessionNumber: number;
  isActive: boolean;
  sections?: Section[];
  userGuid: string;
}

export interface Section {
  sectionGuid: string | undefined;
  sessionGuid: string;
  exerciseGuid: string;
  session: Session;
  order: number;
  bodyPartGuid?: string;
  supersetWithNext?: boolean;
  sectionType: "Strength" | "Crossfit" | "Hypertrophy";
}

export interface CrossfitSection extends Section {
  wodName: string;
  description: string;
}

export interface HypertrophySection extends Section {
  exerciseName: string;
  weight: number;
  reps: number;
  sets: number;
  setResults: string;
}

File: PumpLogApi/Entities/Section.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace PumpLogApi.Entities
{
    [JsonPolymorphic(TypeDiscriminatorPropertyName = "sectionType")]
    [JsonDerivedType(typeof(HypertrophySection), typeDiscriminator: "Hypertrophy")]
    [JsonDerivedType(typeof(CrossfitSection), typeDiscriminator: "Crossfit")]
    public class Section
    {
        public Guid SectionGuid { get; set; }
        public Guid SessionGuid { get; set; }
        [JsonIgnore]
        public Session? Session { get; set; }
        public required Guid ExerciseGuid { get; set; }
        public int Order { get; set; }
        public bool SupersetWithNext { get; set; }

    }

    public class CrossfitSection : Section
    {
        public required string WodName { get; set; }
        public required string Description { get; set; }
    }
    public class HypertrophySection : Section
    {
        public required string ExerciseName { get; set; }
        public decimal Weight { get; set; }
        public int Reps { get; set; }
        public int Sets { get; set; }
        public string SetResults { get; set; } = string.Empty;
    }
}